File: src/main/java/com/cuupa/mailprocessor/services/archive/LocalArchiver.java
package com.cuupa.mailprocessor.services.archive;

import java.io.ByteArrayInputStream;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class LocalArchiver implements FileProtocol {

    @Override
    public void init(String username, String password) {
        // Not implemented
    }

    @Override
    public boolean exists(String path) {
        return Files.exists(Paths.get(path));
    }

    @Override
    public boolean save(String path, byte[] data) {
        try {
            Files.copy(new ByteArrayInputStream(data), Paths.get(path));
            return true;
        } catch (IOException e) {
            return false;
        }
    }

    @Override
    public boolean createDirectory(String path) {
        try {
            Files.createDirectory(Paths.get(path));
            return true;
        } catch (IOException e) {
            return false;
        }
    }

    @Override
    public List<ArchiveResource> list(String path) {
        try {
            return Files.list(Paths.get(path))
                        .map(e -> new ArchiveResource(e.getFileName().toString(), getContentType(e)))
                        .collect(Collectors.toList());
        } catch (IOException e) {
            return new ArrayList<>();
        }
    }

    private String getContentType(Path e) {
        try {
            return Files.probeContentType(e);
        } catch (IOException ioException) {
            return "*/*";
        }
    }

    @Override
    public void close() {
        // Not implemented
    }
}

File: src/main/java/com/cuupa/mailprocessor/services/archive/LocalArchiver.kt
package com.cuupa.mailprocessor.services.archive

import java.io.ByteArrayInputStream
import java.io.IOException
import java.io.InputStream
import java.nio.file.Files
import java.nio.file.Path
import java.nio.file.Paths
import java.util.*
import java.util.stream.Collectors

class LocalArchiver : FileProtocol {

    override fun init(username: String?, password: String?) {
        // Not implemented
    }

    override fun exists(path: String, filename: String): Boolean {
        return Files.exists(Paths.get(path + filename))
    }

    override fun save(path: String, filename: String, data: ByteArray): Boolean {
        return try {
            Files.copy(ByteArrayInputStream(data), Paths.get(path + filename))
            true
        } catch (e: IOException) {
            false
        }
    }

    override fun createDirectory(path: String): Boolean {
        return try {
            Files.createDirectory(Paths.get(path))
            true
        } catch (e: IOException) {
            false
        }
    }

    override fun list(path: String): List<ArchiveResource> {
        return try {
            Files.list(Paths.get(path))
                    .map { e: Path ->
                        ArchiveResource(e.fileName.toString(), getContentType(e))
                    }.collect(Collectors.toList())
        } catch (e: IOException) {
            ArrayList()
        }
    }

    override fun get(name: String, path: String): InputStream {
        TODO("Not yet implemented")
    }

    override fun delete(path: String, filename: String): Boolean {
        TODO("Not yet implemented")
    }

    override fun createDirectories(url: String, path: String): String {
        TODO("Not yet implemented")
    }

    private fun getContentType(e: Path): String {
        return try {
            Files.probeContentType(e)
        } catch (ioException: IOException) {
            "*/*"
        }
    }

    override fun close() {
        // Not implemented
    }
}